internal/handlers/pg/msg_find.go

```diff
@@ -70,6 +70,17 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 		return nil, err
 	}
 
+	maxTimeMS, err := common.GetPositiveNumber(document, "maxTimeMS")
+	if err != nil {
+		return nil, err
+	}
+
+	runCtx, runCancel := context.WithTimeout(ctx, time.Duration(maxTimeMS)*time.Millisecond)
+	defer runCancel()
+	if maxTimeMS != 0 {
+		ctx = runCtx
```

Comment:
If `maxTimeMS` is zero we should not even create a new context that we will not use

Reply:
Fixed